.github/workflows/ci.yml

```diff
@@ -99,8 +104,14 @@ jobs:
           key:
             cache|${{ runner.os }}|${{ env.PY }}|${{ hashFiles('pyproject.toml') }}|${{
             hashFiles('poetry.lock') }}|${{ hashFiles('.pre-commit-config.yaml') }}
-      - run: python -m pip install poetry poetry-dynamic-versioning
-      - run: poetry install
+      - name: Install poetry
+        shell: bash
+        run: |
+          python -m pip install poetry poetry-dynamic-versioning
+          echo "::set-env name=PATH::$HOME/.poetry/bin:$PATH"
+      - run: |
+          poetry run python -m pip install pip -U
+          poetry install
```

Comment:
Again, please revert this.

Reply:
I can revert all the changes except the pip upgrade which is necessary for `poetry install` to work if there is anything more than basic toml syntax used for anything in pyproject.toml, for example dotted paths cause pip to crash.

This is because pip is used inside `poetry install` (I believe there's a plan to change this in future) and the default version of pip uses pytoml for toml parsing which is deprecated and quite limited.